api/src/modules/scenario-interventions/scenario-interventions.service.ts

```diff
@@ -439,4 +408,30 @@ export class ScenarioInterventionsService extends AppBaseService<
 
     return newSourcingLocationData;
   }
+
+  private async replaceCanceledSourcingLocations(
```

Comment:
Minor detail, probably a more concise function name would be createReplacingSourcingLocationsForIntervention()